### PR TITLE
Fix session verification for shares

### LINKED ISSUE #22693: Accessing item with share URL throws SQL error
### Describe the Bug

When trying to access a shared item with its URL `verifySessionJWT` is called with unexpected parameters causing a SQL error along the lines of 

```
Undefined binding(s) detected when compiling FIRST. Undefined column(s): [user] query: select 1 from `directus_sessions` where `token` = ? and `user` = ? and `expires` >= ? limit ?
```

### To Reproduce

- Create a share
- Copy URL
- Open URL in private window

### Directus Version

v10.12.0

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

_No response_

### PR DESCRIPTION
**Pull request by ikovac**
## Scope

What's changed:

- Cast user ID to null if it's undefined when performing knex query.

## Potential Risks / Drawbacks

None

## Review Notes / Questions

None

---

## Current state:

When navigating to the shared URL as an anonymous user, SQL error is thrown:
`ERROR: Undefined binding(s) detected when compiling FIRST. Undefined column(s): [user] query: select 1 from "directus_sess ions" where "token" = ? and "user" = ? and "expires" >= ? limit ?`.

Kapture 2024-06-27 at 12 06 37

## New state:

When navigating to the shared URL as an anonymous user, everything works as expected.

Kapture 2024-06-27 at 12 04 33

Fixes #22693 (together with #22867)

### CODE REVIEW

File: api/src/utils/verify-session-jwt.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export async function verifySessionJWT(payload: DirectusTokenPayload) {
 		.from('directus_sessions')
 		.where({
 			token: payload['session'],
-			user: payload['id'],
+			user: payload['id'] || null,
```
Comment (paescuj) [inline api/src/utils/verify-session-jwt.ts:18]: We should include the share ID in the where clause

```suggestion
			user: payload['id'] || null,
			share: payload['share'] || null,
```

Comment (paescuj) [reply api/src/utils/verify-session-jwt.ts:18]: Note that #22693 is required for that

File: .changeset/kind-olives-drum.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@directus/api': minor
+---
+
+Fix accessing the shared link as an anonymous user
```
Comment (paescuj) [inline .changeset/kind-olives-drum.md:5]:
```suggestion
'@directus/api': patch
---

Fixed session verification for shares
```

File: api/src/utils/verify-session-jwt.ts (hunk 1 above)
Comment (ikovac) [reply api/src/utils/verify-session-jwt.ts:18]: Thank you for the review! It's fixed now :)

**Review by paescuj — APPROVED** [review]

**Pull request merged**